### PR TITLE
Allow multi select on device's image picker where available

### PR DESCRIPTION
**Pull request by mzorz**
With this PR users can insert multiple images from their device in the Media Browser and Editor.

To test:
## Media browser tests

**CASE A: Media Browser, single select, image**
1. go to the Media Browser
2. tap on the `+` icon on the top-right corner of the screen
3. tap `Choose photo from device`
4. tap on any image
5. see the image starts uploading and the notification is presented `1 media file remaining`

**CASE B: Media Browser, multiple select, image** 
1. repeat steps 1-3 in case A
2. long-press one of the images to turn the multiselect mode ON
3. tap on several images (more than 1)
4. tap on “OPEN”
5. see all the selected images get inserted in the Media Browser grid and start uploading. The notification reads `n media files remaining`.

**CASE C: single & multiselect, videos**
1. repeat cases A and B but this time use the `Choose video from device` option.

## Editor tests

**CASE D: Editor: single image, advertising on**
1. clean the app data, so the image optimisation advertising dialog will be shown
2. login and start a new draft
3. tap on the media picker
4. with the media picker up, tap on the left button in the media picker
5. select `Choose photo from device`
6. tap on any image
7. see the image optimisation dialog is shown. Choose either `LEAVE OFF` or turn it on.
8. see the image starts uploading and the notification is presented `1 media file remaining`

**CASE E: Editor: single image, advertising off**
1. start a new draft
2. tap on the media picker
3. with the media picker up, tap on the left button in the media picker
4. select `Choose photo from device`
5. tap on any image
6. see the image gets inserted in the Post  and starts uploading. Also the notification is presented `1 media file remaining`

**CASE F: Editor: multiple images, advertising off**
1. clean the app data, so the image optimisation advertising dialog will be shown
2. login and start a new draft
3. tap on the media picker
4. with the media picker up, tap on the left button in the media picker
5. select `Choose photo from device`
6. use long press on any image to activate multiselect, and then choose several images.
7. tap `OPEN` on the top-right corner of the screen.
8. see the images get inserted in the Post and start uploading and the notification is presented `n media files remaining`

**CASE G: Editor: videos**
1. Repeat cases E and F but tap on `Choose video from device` in steps 4 and 5 respectively.

## Media Notifications tests

**Case H: Media notification & Write Post action**
1. go to the Media Browser
2. tap on the `+` icon on the top-right corner of the screen
3. tap `Choose photo from device`
4. long-press one of the images to turn the multiselect mode ON
5. tap on several images (more than 1)
6. tap on “OPEN”
7. see all the selected images get inserted in the Media Browser grid and start uploading. The notification reads `n media files remaining`.
8. send the app to the background and wait for all uploads to finish.
9. when the progress notification is gone, see the success notification appear, with the `WRITE POST` quick action.
10. tap on the `WRITE POST` quick action
11. observe the Editor is opened and all the newly uploaded images get inserted there.

cc @nbradbury

### CODE REVIEW

**Comment by nbradbury** [comment]
With the Aztec editor, selecting two images to insert into a post inserts two copies of the second one. This doesn't happen with the visual editor.

screenshot_1509630764

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 1]
```diff
@@ -2052,6 +2052,32 @@ private void addMediaLegacyEditor(Uri mediaUri, boolean isVideo) {
         mEditorFragment.appendMediaFile(mediaFile, mediaFile.getFilePath(), mImageLoader);
     }
 
+    private void addMediaItemGroupOrSingleItem(Intent data) {
+        ClipData clipData = data.getClipData();
+        if (clipData != null) {
+            for (int i = 0; i < clipData.getItemCount(); i++) {
+                ClipData.Item item = clipData.getItemAt(i);
+                addMedia(item.getUri(), false);
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: Is there any chance `getUri()` can return null here?

Comment (mzorz) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: yes, but `addMedia` does the null check immediately

**Comment by mzorz** [comment]
> With the Aztec editor, selecting two images to insert into a post inserts two copies of the second one. 

This is super weird, I can't reproduce that here. Tried with 2 images, several images, making these images the very first two, selecting them in left to right order, then right to left... I always get the images I selected. Could you look into it a bit more and see if there's any pattern that lead us to the culprit?

**Comment by nbradbury** [comment]
@mzorz I've tried to repro that on multiple emulators but I can't, and now it no longer happens on the emulator I originally tested with (Nexus 5X API24). I'd like to write this off as a fluke, except I repro'd it several times the first time around. I guess for now we just keep an eye on this?

**Comment by nbradbury** [comment]
:shipit:

**Comment by mzorz** [comment]
> I guess for now we just keep an eye on this?

Sounds good! 🙇

**Pull request merged**